code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -731,6 +737,9 @@
 	if(src.resources < FLOCK_LAY_EGG_COST)
 		boutput(src, "<span class='alert'>Not enough resources (you need [FLOCK_LAY_EGG_COST]).</span>")
 		return
+	if(src.floorrunning)
```

Comment:
What about restrained check?

Reply:
what do you mean?

Reply:
probably better to add that as an extra condition in future, otherwise being handcuffed while trying to lay an egg as a drone would give you a message about floorrunning. Also I can't think why being handcuffed would prevent egg laying.

Reply:
Flockdrones can't actually be handcuffed